lib/saluki-components/src/transforms/host_tags/mod.rs

```diff
@@ -60,12 +60,14 @@ impl SynchronousTransformBuilder for HostTagsConfiguration {
         let context_resolver = ContextResolverBuilder::from_name("host_tags")
             .expect("resolver name is not empty")
             .with_interner_capacity_bytes(context_string_interner_size)
+            .with_idle_context_expiration(Duration::from_secs(self.expected_tags_duration))
```

Comment:
We should ideally set this _lower_, that way infrequent contexts can be cleared out. As is, we would essentially never expire any context up until the point we dropped the entire resolver, so it would just keep growing and growing.

If we set it lower (say thirty seconds? I'm not sure we care deeply about making this configurable), then we can gradually recover some memory while it's active.

Reply:
The part where it checks if the duration elapsed happens in `transform_buffer` which I wasn't sure how to test so I just set `context_resolver` and `host_tags` to `None` for that test. Happy to update if there is a better way to test the 'stop enrichment' part though 🤔 .

Reply:
I think that's fine overall.

The main two way to test this are: set the resolver/tags to `None` and call `enrich_metric` _or_ pause/advance time manually and call `transform_buffer`.

The "if elapsed > expected duration { ... }" bit is straightforward enough that I hesitate to worry too much about _directly_ testing it.